src/main/java/graphql/nadel/SchemaSourceFactory.java

```diff
@@ -0,0 +1,18 @@
+package graphql.nadel;
+
+
+import com.atlassian.braid.Link;
+import com.atlassian.braid.SchemaNamespace;
+import com.atlassian.braid.SchemaSource;
+import graphql.nadel.dsl.ServiceDefinition;
+import graphql.schema.idl.TypeDefinitionRegistry;
+
+import java.util.List;
+
+/**
+ * Factory for schema source.
+ */
+public interface SchemaSourceFactory {
```

Comment:
I am not sure this should be part of the Nadel API: I think we should hide Braid Impl Details from the client